draft-gazdag-x509-hash-sigs.md

```diff
@@ -108,43 +136,64 @@ lists.
-algorithm parameters would allow a private key to sign a virtually unbounded
-number of messages (e.g. 2^60), this is at the cost of a larger signature size
-and longer signing time. Since these algorithms are already known to be secure
-against quantum attacks, and because roots of trust are generally long-lived and
-can take longer to be deployed than end-entity certificates, these signature
-algorithms are more appropriate to be used in root and subordinate CA
-certificates. They are also appropriate in non-interactive contexts such as code
-signing. In particular, there are multi-party IoT ecosystems where publicly
-trusted code signing certificates are useful.
+The private key of HSS, XMSS and XMSS^MT is a finite collection of OTS keys, hence only a limited
+number of messages can be signed and the private key's state must be updated and persisted
+after signing to prevent reuse of OTS keys. Due to this statefulness of the private key
+and the limited number of signatures that can be created, these signature algorithms might
+not be appropriate for use in interactive protocols. While the right selection of algorithm
+parameters would allow a private key to sign a virtually unbounded number of messages (e.g. 2^60),
+this is at the cost of a larger signature size and longer signing time. Since these algorithms
+are already known to be secure against quantum attacks, and because roots of trust are generally
+long-lived and can take longer to be deployed than end-entity certificates, these signature
+algorithms are more appropriate to be used in some use case scenarios.
 
 The private key of SPHINCS+ is a finite but very large collection of FTS keys
 and hence stateless. This typically comes at the cost of larger signatures
 compared to the stateful HBS variants. Thus SPHINCS+ is suitable for more
-use-cases if the signature sizes fit the requirements.
+use cases if the signature sizes fit the requirements.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
 The parameter 'n' is the security parameter, given in bytes. In practice this
 is typically aligned to the standard output length of the hash function in use,
-i.e. either 24, 32 or 64 bytes. The height of a single tree is typically given
-by the parameter 'h'. The number of levels of trees is either called 'L' (HSS)
-or 'd' (XMSS, XMSS^MT, SPHINCS+).
-
-# Subject Public Key Algorithms
+i.e. either 16, 24, 32 or 64 bytes. The height of a single tree is typically
+given by the parameter 'h'. The number of levels of trees is either called 'L'
+(HSS) or 'd' (XMSS, XMSS^MT, SPHINCS+).
+
+# Use Cases of HBS in X.509
+
+As many cryptographic algorithms that are considered to be quantum-resistant,
+HBS have several pros and cons regarding their practical usage. On the positive side
+they are considered to be secure against a classical as well as a quantum adversary,
+a secure instantiation of HBS may always be built as long as a cryptographically
```

Comment:
```suggestion
and a secure instantiation of HBS may always be built as long as a cryptographically
```
@exception42 : I've taken Kaveh's comment to submit a suggestion. Please commit if ok.